controllers/vehicle.go

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/edgarduran/Go-Dispatch-Bootcamp/usecase"
+	"github.com/gorilla/mux"
+)
+
+type vehicleController struct {
+}
+
+func New() vehicleController {
+	return vehicleController{}
+}
+
+func (c vehicleController) GetAllVehicles(w http.ResponseWriter, rtr *http.Request) {
+	vehicles, err := usecase.GetAllVehicles()
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "error getting all vehicles: %v\n", err)
+
+		return
+	}
+
+	jsonData, err := json.Marshal(vehicles)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonData)
+}
+
+func (c vehicleController) GetVehicleById(w http.ResponseWriter, rtr *http.Request) {
+	params := mux.Vars(rtr)
+	id, err := strconv.Atoi(params["id"])
+
+	if err != nil {
```

Comment:
Normally when you obtain an error, the very next line should be the infamous `err != nil ` check. 
```suggestion
	id, err := strconv.Atoi(params["id"])
	if err != nil {
```